spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -1473,9 +1503,371 @@ static void queryWithQueryOptions(DatabaseClient dbClient) {
+      RetryingFuture<OperationSnapshot> pollingFuture = op.getPollingFuture();
+      // Wait for the operation to finish.
+      // isDone will return true if the operation has finished successfully or if it was cancelled
+      // or any other error occurred..
+      while (!pollingFuture.get().isDone()) {
+        System.out.println("Waiting for the cancelled backup operation to finish...");
+        Thread.sleep(TimeUnit.MILLISECONDS.convert(5, TimeUnit.SECONDS));
+      }
+      if (pollingFuture.get().getErrorCode() == null) {
+        // Backup was created before it could be cancelled. Delete the backup.
+        System.out.println("Deleting backup [" + backup.getId() + "]...");
+        backup.delete();
+      } else if (pollingFuture.get().getErrorCode().getCode() == StatusCode.Code.CANCELLED) {
+        System.out.println("Backup operation for [" + backup.getId() + "] successfully cancelled");
+      }
+    } catch (ExecutionException e) {
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+  // [END spanner_cancel_create_backup]
+
+  // [START spanner_list_backup_operations]
+  static void listBackupOperations(InstanceAdminClient instanceAdminClient, DatabaseId databaseId) {
+    Instance instance = instanceAdminClient.getInstance(databaseId.getInstanceId().getInstance());
+    // Get create backup operations for the sample database.
+    String filter =
+        String.format(
+            "(metadata.database:%s) AND "
+                + "(metadata.@type:type.googleapis.com/"
+                + "google.spanner.admin.database.v1.CreateBackupMetadata)",
+            databaseId.getName());
+    Page<Operation> operations = instance.listBackupOperations(Options.filter(filter));
+    for (Operation op : operations.iterateAll()) {
+      try {
+        CreateBackupMetadata metadata = op.getMetadata().unpack(CreateBackupMetadata.class);
+        System.out.println(
+            String.format(
+                "Backup %s on database %s pending: %d%% complete",
+                metadata.getName(),
+                metadata.getDatabase(),
+                metadata.getProgress().getProgressPercent()));
+      } catch (InvalidProtocolBufferException e) {
+        // The returned operation does not contain CreateBackupMetadata.
+        System.err.println(e.getMessage());
+      }
+    }
+  }
+  // [END spanner_list_backup_operations]
+
+  // [START spanner_list_database_operations]
+  static void listDatabaseOperations(
+      InstanceAdminClient instanceAdminClient,
+      DatabaseAdminClient dbAdminClient,
+      DatabaseId databaseId) {
+    Instance instance = instanceAdminClient.getInstance(databaseId.getInstanceId().getInstance());
+    // Get optimize restored database operations for the sample database.
+    String filter = String.format("(name:%s) AND (metadata.@type:type.googleapis.com/"
```

Comment:
In the other samples, we ended up listing all optimize operations so if you could remove the name filter that would be great. Hope it doesn't mess up the tests too much.

Reply:
done